keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1237 @@
+    * max (this option is to specify max usage constraints on the resource)
+    * defaultRequest (this option is to specify default resource request for the named resource)
+    * default (this option is to specify default limit for the named resource)
+    * maxLimitRequestRatio (represents the max burst value of the named resource)
+  
+  default and defaultRequest are not supported by LimitRanger of type `Pod`.
+
+  To incorporate Pod level resource specifications, LimitRanger logic needs to be modfied for Pod and Container types. For PersistentVolumeClaim, 
+  LimitRanger will remain unchanged as PVCs are independent resources and their size is not directly related to the resources consumed by a pod. 
+  Following are the two design options to consider:
+  
+  * **Option 1: Enhance existing `Pod` type**
+      
+      * Add support for default pod request (`defaultRequest`) and limit (`default`) in the `Pod` LimitRange object to allow customers
+      to specify default requests and limits for pods in a namespace. This requires changing the 
+      (validation logic)[https://github.com/kubernetes/kubernetes/blob/9e2075b3c87061d25759b0ad112266f03601afd8/pkg/apis/core/validation/validation.go#L6403]
+      to allow `defaultRequest` and `default` in Pod LimitRange object.
+
+      * Conditional logic
+        
+         * Pod level resource spec is set and `Pod` LimitRanger is applied.
+           If pod level resources are specified, use these values directly against the Pod type LimitRanger. It doesn't matter if container level
+           resource spec is set or not.
+           
+         * Pod level resource spec is set and `Pod` LimitRanger is applied.
+           If pod level resources are not specified, there could be cases:
+             1. Existing workloads that have `Pod` LimitRanger set to validate the constraints at container level aggregates (and not at pod level).
+             2. New workloads that want to use Pod Level resource limits feature and have set `Pod` LimitRanger to validate constraints at Pod Level.
+           
+           If the behavior of `Pod` LimitRanger is modified, it could break existing workloads and also remove the ability for usecase where users 
+           want to have limit range constraints for container level aggregates.         
+
+  * **Option 2: Add a new LimitRanger type PodLevelResources**
+    * Introduce a new type `PodLevelResources` in the LimitRange specifically for defining pod-level resource constraints (min, max, defaultRequest, default, maxLimitRequestRatio).
+  
+    This new type would be independent of the existing Pod and Container LimitRange types with clearer semantics. It maintains backward compatibility as it doesn't modify
+    existing Pod type. User education will be needed to clarify the difference between Pod and PodLevelResources types. Plus this is confusing with Pod type LimitRange.
+
+  * **[Preferred] Option 3: Add a new optional field `Scope` in LimitRangeSpec**
```

Comment:
I strongly prefer using the field that already describers itself as "Pod" if we can. I'd like to understand why that can't work.

Reply:
+1. 

It should work. And since we disallow setting the default values for pods today, opening this up should not break anyone. 

@ndixita to update the section